### PR TITLE
add rollmint docs

### PR DESCRIPTION
**Pull request by jcstein**
- [x] Add Rollmint docs
- [x] add original diagrams
- [x] request review from Rollmint team
- [ ] review Gabriel and Tomasz feedback
- [ ] Request official diagrams from Alina

### CODE REVIEW

**Comment by nashqueue** [comment]
@jcstein Should we review this docs as it is still in draft stage ?

**Comment by jcstein** [comment]
I think it's ready for review @nashqueue and @tzdybal - I imagine there may be a good bit of edits here. I'm also happy to split up this into multiple pages if that makes more sense

File: docs/developers/rollmint.md [hunk 1]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+#### Architecture Overview
+
+- Rollmint aggregator sends block to main chain for DA and Consensus (Celestia,
+Ethereum, Cosmos Hub)
+- If time passes without fraud proof, the block is considered valid
+- Else, if fraud proof, orphan all subsequent blocks
+- With bond on Ethereum, or with longer block times and p2p gossip in
+side-chains subnet, or, liquidity providers
+
+![lazyledger-diagram.png](/img/rollmint/lazyledger-diagram.png)
```
Comment (nashqueue) [inline docs/developers/rollmint.md:88]: This diagram still implies Optimistic Rollups.

**Review by nashqueue — CHANGES_REQUESTED** [review]
I think some of the points will need a revamp as we came to new design decisions since lazyledger.

File: docs/developers/rollmint.md [hunk 2]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+application-agnostic piece of the stack, namely the ABCI Client which defaults
+to `tendermint-core`. We will replace the default ABCI Client with one that
+does not handle Consensus itself. Instead it will only provide the networking
+Layers, and will use the Consensus of the DA Layer. How exactly will be
+described below:
+
+To summarize, if a current SDK-based app roughly follows the following
+architecture:
+
+![diagram1.png](/img/rollmint/diagram1.png)
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:55]: good idea, will leave this unresolved for visibility

File: docs/developers/rollmint.md [hunk 3]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+When implementing a Cosmos-SDK based Rollmint chain, this overview changes to:
+
+![diagram2.png](/img/rollmint/diagram2.png)
+
+Note that from the point of view of an application developer, nothing changes.
+Rollmint has the exact same developer experience as the usual Cosmos-SDK with
+Tendermint provides: they still just have to define their business /
+state-machine logic. The Rollmint-specific ABCI Client is stripped down to only
+handle the network Layer but it also embeds a light node of the DA and
+Consensus Layer chain used and takes care of submitting the block to the DA
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:66]: yep, good thinking, adding to next commit

File: docs/developers/rollmint.md [hunk 4]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+
+As with Tendermint the state-machine and Rollmint interact with each other via
+ABCI.
+
+#### Definitions
+
+- **Rollmint Block Time** - configurable duration after which a new block of
+the (Rollmint chain) is produced.
+- **Grace Period** - configurable duration between block creation and block
+finality. It’s the time window for fraud proof submission.
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:77]: I think differentiating makes the most sense, but I don't know that there is guidance for anything other than a "pessimistic" rollup at the moment?

File: docs/developers/rollmint.md [hunk 5]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+
+#### Architecture Overview
+
+- Rollmint aggregator sends block to main chain for DA and Consensus (Celestia,
+Ethereum, Cosmos Hub)
+- If time passes without fraud proof, the block is considered valid
+- Else, if fraud proof, orphan all subsequent blocks
+- With bond on Ethereum, or with longer block times and p2p gossip in
+side-chains subnet, or, liquidity providers
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:86]: me neither. @tzdybal ?

File: docs/developers/rollmint.md (hunk 1 above)
Comment (jcstein) [reply docs/developers/rollmint.md:88]: yes, noting another for Alina to reproduce

File: docs/developers/rollmint.md [hunk 6]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+
+Full nodes are a crucial part of the networks, because they are responsible for
+producing blocks and fraud proofs. They also create a link between the Rollmint
+network and the DA and Consensus Layer, by pushing aggregates to the DA and
+Consensus Layer.
+
+##### Light Nodes
+
+Light nodes are the main producer of transactions in the Rollmint network.
+They participate in gossiping of both transactions and fraud proofs. Light
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:100]: ✅

File: docs/developers/rollmint.md [hunk 7]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+nodes may only request or store a subset of the state, just to ensure that they
+can execute rollback.
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:102]: cc @tzdybal

File: docs/developers/rollmint.md [hunk 8]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+
+<!-- markdownlint-disable MD013 -->
+##### Rather Long Block Times, so Fraud Proofs can be Gossiped before the Block is Complete
+<!-- markdownlint-enable MD013 -->
+
+We need to experiment with block times and grace period values to determine what
+works best. Our first assumption is that Rollmint chain is not targeting short
+block-times, something like 60 seconds sound reasonable. The grace period should
+be significantly shorter, like 30 seconds. In such a scenario, we can assure
+that no more than one block at a time will be rolled back.
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:161]: sounds like this has changed and we should change it. can you confirm @tzdybal ?

File: docs/developers/rollmint.md [hunk 9]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+places seem managable and having a clear interface between state-machine and the
+DA and Consensus Layer makes it possible to replace Tendermint with another
+ABCI Client.
+
+As we want to change as little as possible inside the SDK we will surely have to
+stick to the vanilla ABCI interface. That said, some of the properties we also
+want for the Rollmint chains, e.g., that the intermediate state roots and the
+final state root (e.g. app-hash) of a block match the state transitions implied
+by the Transactions included in that very same block require changes in the ABCI
+itself. While ABCI++ is not part of this project’s scope, we are
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:188]: 👀 what do you think @liamsi ?

File: docs/developers/rollmint.md [hunk 10]
```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
...
+  // This should create a transaction which (potentially)
+  // triggers a state transition in the DA and Consensus Layer
+  SubmitBlock(block types.Block) ResultSubmitBlock
+}
+```
+
+The `SubmitBlock` method will be called on an aggregator node before Rollmint
+tells the application (via ABCI) to commit the state (via the Commit method
+which might be renamed to `FinalizeBlock` - see discussion on
+[ABCI++ PR](https://github.com/tendermint/spec/pull/254)). Only if `SubmitBlock`
```
[the remark this answers is not in the document]
Comment (jcstein) [reply docs/developers/rollmint.md:270]: yes, that is what this is/was initially

File: docs/developers/rollmint.md [hunk 11]
```diff
@@ -11,14 +11,14 @@ sidebar_label: Rollmint
 rollups to deploy on top of Celestia.
 
 It is built by replacing Tendermint, the Cosmos-SDK
-consensus layer, with a drop-in replacement that
-communicates directly with Celestia's Data Availability layer.
+Consensus Layer, with a drop-in replacement that
+communicates directly with Celestia's Data Availability (DA) and Consensus Layer.
```
Comment (adlerjohn) [inline docs/developers/rollmint.md:15]: Hmmm IIRC @YazzyYaz was saying that docusaurus enforces a line character limit

**Pull request closed without merging**